packages/integrations/mdx/src/remark-images-to-component.ts

```diff
@@ -24,7 +24,7 @@ export function remarkImageToComponent() {
 				// If we haven't already imported this image, add an import statement
 				if (!importName) {
 					importName = `__${importedImages.size}_${node.url.replace(/\W/g, '_')}__`;
-
+					const prefixedNodeUrl = (node.url.includes('/') ? '' : './') + node.url;
```

Comment:
This does not address my comment: https://github.com/withastro/astro/pull/10705#discussion_r1556036393

> This works as expected with prefixedNodeUrl as verified using tests.

The tests doesn't seem to cover aliases, but that's something we indeed support and we can't break it. We can add the alias test if needed here.

Reply:
@bluwy

https://github.com/withastro/astro/blob/1f8661ce4468c415341327adcbf1c4761318ab3b/packages/integrations/mdx/test/fixtures/mdx-images/src/pages/index.mdx?plain=1#L11 we're already testing aliases, aren't we?

Reply:
due to `foo/img.png` having a `/` in it I think this wouldn't prepend `./` onto it

I didn't consider aliasing directly to a file, that is a bit of a problem honestly 

Would this be affecting https://github.com/withastro/astro/pull/9755 as well @bluwy

Reply:
You are totally correct, @bluwy.

The aliases work but still required to be resolved given the unknown alias's string.

I've added a relative image like you mentioned, including a test for it.

Reply:
I am still figuring out a way to invoke rollup's resolve in the tree visitor function.

Reply:
@OliverSpeir Yeah it looks like that has the same issue as this too. In that case, we need to fix for both MD and MDX. Thinking about it, maybe the better fix altogether is a Vite plugin something like this:

```js
{
  async resolveId(id, importer, opts) {
    if (importer.endsWith('.md') && id[0] !== '/') {
      let resolved = await this.resolve(id, importer, opts)
      if (!resolved) {
        resolved = await this.resolve('./' + id, importer, opts)
      }
      return resolved
  }
}
```

`vite-plugin-markdown`, `@astrojs/mdx`, and `@astrojs/markdoc` can each implement this themselves.



> I am still figuring out a way to invoke rollup's resolve in the tree visitor function.

I think we should try to avoid this if possible so we have a proper separation of handling/compiling md/mdx.

Reply:
Sure, updated the code.